### ISSUE TITLE
fix: "Approve" on swap UI is bugged

### ISSUE THREAD

**Issue by JULl4N** [ZigZagExchange/mammoth_UI#51]
1) Wrong token gets approved

Swap 
`FROM` FC
`to` TUSDC
Click approve
TUSDC gets approved, while FC should be getting approved
![Screenshot 2022-06-11 at 07 46 46](https://user-images.githubusercontent.com/103177338/173174976-930b4938-e16a-4ea5-b339-34e1dffbabdf.png)

2) Now when I click approve it keeps approving FC, but it does not recognise that FC already has been approved. The approve button stays. After successfully approving FC for the second time:
![Screenshot 2022-06-11 at 07 54 38](https://user-images.githubusercontent.com/103177338/173175219-caf4368c-c765-4777-b80c-a9604863d08f.png)

So in short what I did:
1. Swap UI on FC -> TUSDC
2. Click approve: it approves TUSDC
3. Click approve again, it approves FC
4. Click approve again, I can sign a tx again to approve FC (but nothing happens/changes after I approve twice, I lose tx fee and no successful pop up is shown)

**Comment by 0xtrooper** [comment]
fixed with #70
